src/main/java/com/efficio/fieldbook/web/trial/controller/CreateTrialController.java

```diff
@@ -182,6 +185,17 @@ void excludeTreatmentFactorVariables(final Workbook trialWorkbook) {
 		trialWorkbook.setTreatmentFactors(new ArrayList<>());
 	}
 
+	void excludeObsoleteVariables(final Workbook trialWorkbook) {
+		final Set<Integer> nonObsoleteVariables = this.fieldbookMiddlewareService.getVariableIdsByObsoleteFilter(true);
```

Comment:
Instead of creating a new method to get all the obsolete variables, you can use org.generationcp.middleware.manager.ontology.OntologyVariableDataManagerImpl#getWithFilter, and filter the non obsolete variables. Then you can remove the variables that were not returned by the filter.